tutorials/RandomNumberGeneration/Tests.qs

```diff
@@ -17,65 +17,108 @@ namespace Quantum.Kata.RandomNumberGeneration {
     open Microsoft.Quantum.Random;
 
     // Exercise 1.
-    @Test("QuantumSimulator")
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T1_RandomBit () : Unit {
-        Message("Testing...");
-        CheckFlatDistribution(RandomBit_Wrapper, 1, 0.4, 0.6, 1000, 450);
+        Message("Testing One RandomBitGeneration...");
+        RetryCheckUniformDistribution(RandomBit_Wrapper, 0, 1, 1000, 3);
+        Message("One random bit successfully generated...");
     }
 
-    operation RandomBit_Wrapper (throwaway: Int) : Int {
+    operation RandomBit_Wrapper (throwawayMin: Int, throwawayMax: Int) : Int {
         return RandomBit();
     }
 
 
     // Exercise 2.
-    @Test("QuantumSimulator")
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T2_RandomTwoBits () : Unit {
-        Message("Testing...");
-        CheckFlatDistribution(RandomTwoBits_Wrapper, 2, 1.4, 1.6, 1000, 200);
+        Message("Testing Two RandomBitGeneration...");
+        RetryCheckUniformDistribution(RandomTwoBits_Wrapper, 0, 3, 1000, 3);
+        Message("Two random bits successfully generated...");
     }
 
-    operation RandomTwoBits_Wrapper (throwaway: Int) : Int {
+    operation RandomTwoBits_Wrapper (throwawayMin: Int, throwawayMax: Int) : Int {
         return RandomTwoBits();
     }
     
 
     // Exercise 3.
-    @Test("QuantumSimulator")
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T3_RandomNBits () : Unit {
         Message("Testing N = 1...");
-        CheckFlatDistribution(RandomNBits, 1, 0.4, 0.6, 1000, 450);
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 1, 1000, 3);
+        Message("Test Passed Succesfully for N=1 \n");
+
         Message("Testing N = 2...");
-        CheckFlatDistribution(RandomNBits, 2, 1.4, 1.6, 1000, 200);
-        Message("Testing N = 3...");
-        CheckFlatDistribution(RandomNBits, 3, 3.3, 3.7, 1000, 90);
-        Message("Testing N = 10...");
-        CheckFlatDistribution(RandomNBits, 10, 461.0, 563.0, 1000, 0);
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 3, 1000, 3);
+        Message("Test Passed Succesfully for N=2 \n");
+
+        Message("\n Testing N = 3...");
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 7, 1000, 3);
+        Message("Test Passed Succesfully for N=3 \n");
+
+        Message("\n Testing N = 10...");
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 1023, 1000, 3);
+        Message("Test Passed Succesfully for N=10 \n");
+    }
+
+    operation RandomNBits_Wrapper (min: Int, max: Int) : Int {
+        // For N bit random number : min = 0, max = 2^N - 1
+        let N = Ceiling( Lg(IntAsDouble(max+1)) );
+        return RandomNBits(N);
+    }
+
+    operation RetryCheckUniformDistribution( fTest: ((Int,Int)=>Int), min: Int, max: Int, nRuns: Int, numRetries: Int) : Unit {
```

Comment:
In all cases the number of retries is 3; would it make sense to remove it as a parameter and to introduce a constant inside this operation?

Reply:
Removed `numRetries` as a parameter since we are going for retry policy of 3 for every operation. Also, I did the same change for `RetryCheckXPercentZero`